functions/go/set-namespace/main.go

```diff
@@ -7,6 +7,43 @@ import (
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
+var notNamespaceableKinds = []string{
```

Comment:
Can you export the list and isNamespaceableKind function from https://github.com/kubernetes-sigs/kustomize/blob/master/api/resid/gvk.go and use it here?

Reply:
`notNamespaceableKinds` from https://github.com/kubernetes-sigs/kustomize/blob/master/api/resid/gvk.go is internal, and can only be used within the same package.
https://github.com/kubernetes-sigs/kustomize/blob/master/api/resid/gvk.go#L195 is a method on a structure, which is more than what we need here.

Reply:
The main motivation is to keep one source of truth list for `notNamespaceableKinds`. If we move the `isNamespaceableKind` implementation to some new file like kustomize/api/resid/util.go then it can use kustomize's `notNamespaceableKinds` and we don't have to maintain duplicate lists.

Reply:
Make sense. Created https://github.com/kubernetes-sigs/kustomize/pull/2857 for this.

Reply:
https://github.com/kubernetes-sigs/kustomize/pull/2859 is merged.
I updated this PR. PTAL.